android_core_components/src/org/ros/android/RosActivity.java

```diff
@@ -202,15 +204,20 @@ protected void bindNodeMainExecutorService() {
     intent.putExtra(NodeMainExecutorService.EXTRA_NOTIFICATION_TITLE, notificationTitle);
     startService(intent);
     Preconditions.checkState(
-        bindService(intent, nodeMainExecutorServiceConnection, BIND_AUTO_CREATE),
-        "Failed to bind NodeMainExecutorService.");
+            bindService(intent, nodeMainExecutorServiceConnection, BIND_AUTO_CREATE),
+            "Failed to bind NodeMainExecutorService.");
   }
 
   @Override
   protected void onDestroy() {
+    finishActivity(MASTER_CHOOSER_REQUEST_CODE);
     unbindService(nodeMainExecutorServiceConnection);
     nodeMainExecutorService.
             removeListener(nodeMainExecutorServiceConnection.getServiceListener());
+    if (!shutdownSignalReceived) {
+      //shutdown if RosActivity is not finished by a shutdown signal
+      nodeMainExecutorService.shutdown();
```

Comment:
```suggestion
      nodeMainExecutorService.forceShutdown();
```

With `shutdown`, a popup asks for confirmation when the app is swiped. This is a bit annoying, and if the user clicks "cancel", the window is closed but the app still will run in the background until it is shut down tapping the notification.
Changing this with `forceShutdown` addresses this tiny issue.